src_tracer/database.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import sqlite3
+import os
+
+from typing import List, Tuple
+
+log = logging.getLogger(__name__)
+
+class Database:
+
+    def __init__(self, store_dir, name="function_database.db", path=None):
+        if path:
+            self.connection = sqlite3.connect(path)
+        else:
+            db_path = os.path.join(store_dir, name)
+            self.connection = sqlite3.connect(db_path)
+
+    def get_name(self, func_num):
+        cursor = self.connection.cursor()
+        cursor.execute("SELECT name FROM function_list WHERE rowid=?", (func_num,))
+        name = cursor.fetchone()
+        if name is None:
+            log.error("there is no name (entry) for the func_num (%d)", func_num)
+            return None
+        cursor.close()
+        return name[0]
+
+    def get_numbers(self, func_name):
+        cursor = self.connection.cursor()
+        cursor.execute("SELECT rowid FROM function_list WHERE name=?", (func_name,))
+        func_nums = [func_num[0] for func_num in cursor.fetchall()]
+        cursor.close()
+        return func_nums
+
+    def create_table(self, table_name: str, colum_name_type: List[Tuple[str, str]], key: List[str]):
+        cursor = self.connection.cursor()
+        columns = ", ".join([" ".join(col) for col in colum_name_type])
+        key_str = ", ".join(key)
+        cursor.execute(f'''
+                       CREATE TABLE IF NOT EXISTS {table_name}
+                            ( {columns}, PRIMARY KEY ({key_str}) )
+                       ''')
+        cursor.close()
+
+    def insert_to_table(self, table_name: str, value: List[Tuple[str, ]]):
+        cursor = self.connection.cursor()
+        column_names = ",".join([col[0] for col in value])
+        column_value = tuple([col[1] for col in value])
+        value_query = ",".join("?" * len(value))
+        sql = f'''INSERT INTO {table_name}
+                ({column_names})
+                VALUES({value_query})
+                ON CONFLICT DO NOTHING'''
+        cursor.execute(sql, column_value)
+        cursor.close()
+        self.connection.commit()
```

Comment:
* remove `table_name` argument (directly use `'function_list'` instead)
* `file`, `line`, `name` should be arguments of the function
So the new signature would look like this:
```Python
def insert_to_table(self, file, line, name):
```